app/controllers/spells_controller.rb

```diff
@@ -0,0 +1,16 @@
+class SpellsController < ApplicationController
+  def index
+    if params[:class_filter]
+      spell_query = FilterBuilder.new({klass: Spell}, params)
+      @spells = Kaminari.paginate_array(spell_query.perform).page(params[:page]).per(50)
```

Comment:
Based on how I built the FilterBuilder, line 4 initiates the filter model. The perform method on line line 5 performs the query and returns it so that the query can be rendered by Kaminari. Currently Kaminari is paginating the entire spell index and also when the index is filtered. So when you filter by bard it will paginate all bard spells to be displayed. The syntax is a bit different for this filter query because the custom SQL I needed to run the query is returned as an array, not an ActiveRecord relation. Line 7 is standard pagination syntax.

Kaminari basically works by splitting the query into groups, and based on the `params[:page]` it will display the corresponding query subset. It can be customized through `config/initializers/kaminari_config.rb`